tests/test_pipeline.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e 
```

Comment:
Since this is bash, I suggest we go full paranoid and use the [complete](https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/) `set -Eeuxo pipefail` configuration.